src/tests/ftest/util/ior_utils.py

```diff
@@ -1,186 +1,297 @@
+        self.daos_svcl = IorParam("--daos.svcl {}")             # pool SVCL
+        self.daos_cont = IorParam("--daos.cont {}")             # cont uuid
+        self.daos_destroy = IorParam("--daos.destroy", True)    # destroy cont
+        self.daos_group = IorParam("--daos.group {}")           # server group
+        self.daos_chunk = IorParam("--daos.chunk_size {}", 1048576)
+        self.daos_oclass = IorParam("--daos.oclass {}")         # object class
+
+    def __str__(self):
+        """Return a IorCommand object as a string.
+
+        Returns:
+            str: the ior command with all the defined parameters
+
+        """
+        params = []
+        for value in self.__dict__.values():
+            value_str = str(value)
+            if value_str != "":
+                params.append(value_str)
+        return " ".join(["ior"] + sorted(params))
+
+    def set_params(self, test, path="/run/ior/*"):
+        """Set values for all of the ior command params using a yaml file.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace. Defaults to "/run/ior/*".
+
+        """
+        for name, ior_param in self.__dict__.items():
+            ior_param.set_yaml_value(name, test, path)
+
+    def get_launch_command(self, basepath, processes, hostfile, runpath=None):
+        """Get the process launch command used to run IOR.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+
+        Raises:
+            IorFailed: if an error occured building the IOR command
+
+        """
+        with open(os.path.join(basepath, ".build_vars.json")) as afile:
+            build_paths = json.load(afile)
+        attach_info_path = os.path.join(basepath, "install/tmp")
+
+        # # Verify required params are defined
+        # for name in ("daos_pool", "daos_cont", "daos_svcl"):
+        #     value = getattr(self, name)
+        #     if str(value) == "":
+        #         raise IorFailed("Missing required {} param".format(name))
+
+        if self.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": attach_info_path,
+                "DAOS_POOL": str(self.daos_pool),
+                "MPI_LIB": "",
+                "DAOS_SVCL": str(self.daos_svcl),
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            export_cmd = [
+                "export {}={}".format(key, val) for key, val in env.items()]
+            mpirun_cmd = [
+               os.path.join(
+                   runpath if runpath else build_paths[
+                       "OMPI_PREFIX"], "bin/mpirun"), "-np {}".format(
+                           processes), "--hostfile {}".format(hostfile),
+            ]
+            command = " ".join(mpirun_cmd + [self.__str__()])
+            command = "; ".join(export_cmd + command)
```

Comment:
this should be command = "; ".join(export_cmd + [command])
or else it will return TyepError